internal/pkg/service/common/etcdop/op/txn.go

```diff
@@ -155,17 +164,29 @@ func (v *TxnOp[R]) lowLevelTxn(ctx context.Context) (*lowLevelTxn[R], error) {
 	out.ifs = make([]etcd.Cmp, len(v.ifs))
 	copy(out.ifs, v.ifs)
 
-	// Map THEN operations
+	// Map Then and ThenTxn operations
 	for i, op := range v.thenOps {
 		// Create low-level operation
-		if lowLevel, err := op.Op(ctx); err == nil {
+		if lowLevel, err := op.Op(ctx); err != nil {
+			errs.Append(errors.PrefixErrorf(err, "cannot create operation [then][%d]", i))
+		} else if lowLevel.Op.IsTxn() {
+			errs.Append(errors.Errorf(`cannot create operation [then][%d]: operation "%T" is a transaction, use ThenTxn, not Then`, i, lowLevel.Op))
```

Comment:
Sure, it would be nice to check it sooner.

But to create a low level operation you need (in some cases) the context, which we don't want here.
- Composing and invoking phases are strictly separated.

I could check the *TxnOp type there, which will cover 99% of cases.
- I will do it now.

But you will still be able to do your own high-level operation, 
e.g. AcquireLock, which is implemented using a low-level transaction.